### ISSUE TITLE
Merge igosoft-patches

### ISSUE THREAD

**Issue by akreuzkamp** [qmlweb/qmlweb#62]
See branch igosoft-patches.

These patches contain a Loader class, childrenRect and some small fixes.

Unfortunately everything is in one commit. Probably needs some splitting up and needs to be rebased on top of current master.
See also https://phabricator.kde.org/D623

/cc @igosoft Maybe you can set hand to it yourself, you know your patches best :)

**Comment by ChALkeR** [comment]
@igosoft, do you have these separate somewhere, or was this initially one huge change?

This has to be split somehow. I can start working on that.
I will keep two additional branches — `igosoft-patches-split` (with the same changes as here and zero diff) and `igosoft-patches-split-rebased` (some of the splitted changes cherry-picked and rebased against master).

Note that there already is an initial `Loader` implementation, but this one looks better to me.
@igosoft, in the mean time, please leave a comment in #52.

**Comment by ChALkeR** [comment]
https://github.com/qmlweb/qmlweb/commits/igosoft-patches — original.
https://github.com/qmlweb/qmlweb/commits/igosoft-patches-split — split.

**Comment by ChALkeR** [comment]
This is a work in progress.

**Comment by ChALkeR** [comment]
@igosoft I'm reverting code moving and your microoptimizations, those are mostly useless and will make merges harder.

Also, there is a mistake here, for example: https://github.com/qmlweb/qmlweb/commit/600e3bd5697aae721a1e7f5f01671719d9bdddd0#diff-de79c85e70cebf990f54694973fa1f62R11990.

But the changes look great overall, thanks! I appreciate the Loader especially =), hope to test it soon.

**Comment by ChALkeR** [comment]
@igosoft, Optimizations (if needed) and code cleanup will be redone after merging everything.

**Comment by ChALkeR** [comment]
Status update: the code splitting is half done, will continue soon.

**Comment by igosoft** [comment]
No. I don't have them separate. At that time I was playing with qmlqweb and done many mistakes. ChALkeR please continue to merge my changes. In case of any questions just let me know.

**Comment by ChALkeR** [comment]
Still working on this, about half-done splitting.

**Comment by henrikrudstrom** [comment]
@igosoft, @ChALkeR would be great to get this one landed, anything i can do to help?

**Comment by igosoft** [comment]
The work is in progress but it depands on @ChALkeR how and if we can help here.

**Comment by ChALkeR** [comment]
See #121 (not ready yet).

**Comment by ChALkeR** [comment]
Per #158, this is blocked by #17.

**Comment by ChALkeR** [comment]
Ok, not blocked by #17 anymore, as we sorted out all the changes done in #17.

**Comment by ChALkeR** [comment]
Superseded by #223.
